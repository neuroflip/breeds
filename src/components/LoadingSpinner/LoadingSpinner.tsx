
import styles from './styles/loadingSpinner.module.css'

const LoadingSpinner = () => {
  return <div className={ styles.loadingSpinner } role="progressbar" aria-label="Content loading" aria-busy="true" aria-live="polite"></div>
}

export default LoadingSpinner