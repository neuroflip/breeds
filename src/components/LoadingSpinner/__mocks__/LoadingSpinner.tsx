const LoadingSpinner = () => {
  return <>LoadingSpinner</>
}

export default LoadingSpinner