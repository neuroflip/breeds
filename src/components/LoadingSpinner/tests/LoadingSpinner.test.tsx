import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import LoadingSpinner from '../LoadingSpinner';

describe('LoadingSpinner', () => {
  test('initial render', () => {
    const { container } = render(<LoadingSpinner />);

    const loadingSpinnerElement = screen.getByRole('progressbar');

    expect(loadingSpinnerElement).toBeInTheDocument();
    expect(loadingSpinnerElement.className).toBe('loadingSpinner');
    expect(loadingSpinnerElement.getAttribute('aria-busy')).toBeTruthy();
    expect(loadingSpinnerElement.getAttribute('aria-live')).toBe('polite');

    //snapshot
    expect(container).toMatchSnapshot()
  });
});