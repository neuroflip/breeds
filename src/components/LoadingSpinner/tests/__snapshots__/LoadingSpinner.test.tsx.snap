// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSpinner > initial render 1`] = `
<div>
  <div
    aria-busy="true"
    aria-live="polite"
    class="loadingSpinner"
    role="progressbar"
  />
</div>
`;
